from PyQt5 import QtCore, QtGui, QtWidgets
import popplerqt5


class Ui_Schenky(object):
    def setupUi(self, Schenky):
        Schenky.setObjectName("Schenky")
        Schenky.resize(910, 591)
        self.centralwidget = QtWidgets.QWidget(Schenky)
        self.centralwidget.setObjectName("centralwidget")
        self.toolBox = QtWidgets.QToolBox(self.centralwidget)
        self.toolBox.setGeometry(QtCore.QRect(800, 20, 100, 211))
        self.toolBox.setObjectName("toolBox")
        self.page = QtWidgets.QWidget()
        self.page.setGeometry(QtCore.QRect(0, 0, 100, 149))
        self.page.setObjectName("page")
        self.slur_button = QtWidgets.QPushButton(self.page)
        self.slur_button.setGeometry(QtCore.QRect(0, 0, 100, 30))
        self.slur_button.setObjectName("slur_button")
        self.unfold_button = QtWidgets.QPushButton(self.page)
        self.unfold_button.setGeometry(QtCore.QRect(0, 40, 100, 30))
        self.unfold_button.setObjectName("unfold_button")
        self.reach_button = QtWidgets.QPushButton(self.page)
        self.reach_button.setGeometry(QtCore.QRect(0, 80, 100, 30))
        self.reach_button.setObjectName("reach_button")
        self.toolBox.addItem(self.page, "")
        self.page_2 = QtWidgets.QWidget()
        self.page_2.setGeometry(QtCore.QRect(0, 0, 100, 149))
        self.page_2.setObjectName("page_2")
        self.urlinie_button = QtWidgets.QPushButton(self.page_2)
        self.urlinie_button.setGeometry(QtCore.QRect(0, 0, 100, 30))
        self.urlinie_button.setObjectName("urlinie_button")
        self.inner_button = QtWidgets.QPushButton(self.page_2)
        self.inner_button.setGeometry(QtCore.QRect(0, 40, 100, 30))
        self.inner_button.setObjectName("inner_button")
        self.toolBox.addItem(self.page_2, "")
        self.key_label = QtWidgets.QLabel(self.centralwidget)
        self.key_label.setGeometry(QtCore.QRect(10, 10, 59, 51))
        font = QtGui.QFont()
        font.setPointSize(11)
        self.key_label.setFont(font)
        self.key_label.setAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.key_label.setObjectName("key_label")
        self.key_combo = QtWidgets.QComboBox(self.centralwidget)
        self.key_combo.setGeometry(QtCore.QRect(90, 20, 121, 31))
        self.key_combo.setObjectName("key_combo")
        self.formLayoutWidget = QtWidgets.QWidget(self.centralwidget)
        self.formLayoutWidget.setGeometry(QtCore.QRect(30, 70, 741, 211))
        self.formLayoutWidget.setObjectName("formLayoutWidget")
        self.note_input = QtWidgets.QFormLayout(self.formLayoutWidget)
        self.note_input.setLabelAlignment(QtCore.Qt.AlignRight|QtCore.Qt.AlignTrailing|QtCore.Qt.AlignVCenter)
        self.note_input.setContentsMargins(0, 0, 0, 0)
        self.note_input.setObjectName("note_input")
        self.treble_input = QtWidgets.QTextEdit(self.formLayoutWidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.treble_input.sizePolicy().hasHeightForWidth())
        self.treble_input.setSizePolicy(sizePolicy)
        self.treble_input.setMaximumSize(QtCore.QSize(16777215, 91))
        self.treble_input.setLineWrapMode(QtWidgets.QTextEdit.NoWrap)
        self.treble_input.setObjectName("treble_input")
        self.note_input.setWidget(0, QtWidgets.QFormLayout.FieldRole, self.treble_input)
        self.bass_label = QtWidgets.QLabel(self.formLayoutWidget)
        font = QtGui.QFont()
        font.setPointSize(11)
        self.bass_label.setFont(font)
        self.bass_label.setObjectName("bass_label")
        self.note_input.setWidget(1, QtWidgets.QFormLayout.LabelRole, self.bass_label)
        self.bass_input = QtWidgets.QTextEdit(self.formLayoutWidget)
        sizePolicy = QtWidgets.QSizePolicy(QtWidgets.QSizePolicy.Expanding, QtWidgets.QSizePolicy.Minimum)
        sizePolicy.setHorizontalStretch(0)
        sizePolicy.setVerticalStretch(0)
        sizePolicy.setHeightForWidth(self.bass_input.sizePolicy().hasHeightForWidth())
        self.bass_input.setSizePolicy(sizePolicy)
        self.bass_input.setMaximumSize(QtCore.QSize(16777215, 91))
        self.bass_input.setLineWrapMode(QtWidgets.QTextEdit.NoWrap)
        self.bass_input.setObjectName("bass_input")
        self.note_input.setWidget(1, QtWidgets.QFormLayout.FieldRole, self.bass_input)
        self.treble_label = QtWidgets.QLabel(self.formLayoutWidget)
        font = QtGui.QFont()
        font.setPointSize(11)
        self.treble_label.setFont(font)
        self.treble_label.setObjectName("treble_label")
        self.note_input.setWidget(0, QtWidgets.QFormLayout.LabelRole, self.treble_label)
        self.horizontal_scroll = QtWidgets.QScrollBar(self.formLayoutWidget)
        self.horizontal_scroll.setOrientation(QtCore.Qt.Horizontal)
        self.horizontal_scroll.setObjectName("horizontal_scroll")
        self.note_input.setWidget(2, QtWidgets.QFormLayout.FieldRole, self.horizontal_scroll)
        self.line = QtWidgets.QFrame(self.centralwidget)
        self.line.setGeometry(QtCore.QRect(30, 290, 741, 16))
        self.line.setFrameShape(QtWidgets.QFrame.HLine)
        self.line.setFrameShadow(QtWidgets.QFrame.Sunken)
        self.line.setObjectName("line")
        self.image_label = QtWidgets.QLabel(self.centralwidget)
        self.image_label.setGeometry(QtCore.QRect(30, 320, 741, 191))
        self.image_label.setText("")
        self.image_label.setObjectName("image_label")
        Schenky.setCentralWidget(self.centralwidget)
        self.menubar = QtWidgets.QMenuBar(Schenky)
        self.menubar.setGeometry(QtCore.QRect(0, 0, 910, 22))
        self.menubar.setObjectName("menubar")
        self.menuFile = QtWidgets.QMenu(self.menubar)
        self.menuFile.setObjectName("menuFile")
        self.menuEdit = QtWidgets.QMenu(self.menubar)
        self.menuEdit.setObjectName("menuEdit")
        Schenky.setMenuBar(self.menubar)
        self.statusbar = QtWidgets.QStatusBar(Schenky)
        self.statusbar.setObjectName("statusbar")
        Schenky.setStatusBar(self.statusbar)
        self.actionNew = QtWidgets.QAction(Schenky)
        self.actionNew.setObjectName("actionNew")
        self.actionOpen = QtWidgets.QAction(Schenky)
        self.actionOpen.setObjectName("actionOpen")
        self.actionSave = QtWidgets.QAction(Schenky)
        self.actionSave.setObjectName("actionSave")
        self.actionExit = QtWidgets.QAction(Schenky)
        self.actionExit.setObjectName("actionExit")
        self.actionCopy = QtWidgets.QAction(Schenky)
        self.actionCopy.setObjectName("actionCopy")
        self.actionCut = QtWidgets.QAction(Schenky)
        self.actionCut.setObjectName("actionCut")
        self.actionPaste = QtWidgets.QAction(Schenky)
        self.actionPaste.setObjectName("actionPaste")
        self.menuFile.addAction(self.actionNew)
        self.menuFile.addAction(self.actionOpen)
        self.menuFile.addAction(self.actionSave)
        self.menuFile.addSeparator()
        self.menuFile.addAction(self.actionExit)
        self.menuEdit.addAction(self.actionCopy)
        self.menuEdit.addAction(self.actionCut)
        self.menuEdit.addAction(self.actionPaste)
        self.menubar.addAction(self.menuFile.menuAction())
        self.menubar.addAction(self.menuEdit.menuAction())

        self.retranslateUi(Schenky)
        self.toolBox.setCurrentIndex(1)
        QtCore.QMetaObject.connectSlotsByName(Schenky)

    def retranslateUi(self, Schenky):
        _translate = QtCore.QCoreApplication.translate
        Schenky.setWindowTitle(_translate("Schenky", "Schenky"))
        self.slur_button.setText(_translate("Schenky", "Slur"))
        self.unfold_button.setText(_translate("Schenky", "Unfolding"))
        self.reach_button.setText(_translate("Schenky", "Reaching-over"))
        self.toolBox.setItemText(self.toolBox.indexOf(self.page), _translate("Schenky", "Elements"))
        self.urlinie_button.setText(_translate("Schenky", "Urlinie"))
        self.inner_button.setText(_translate("Schenky", "Inner Beam"))
        self.toolBox.setItemText(self.toolBox.indexOf(self.page_2), _translate("Schenky", "Line"))
        self.key_label.setText(_translate("Schenky", "Key:"))
        self.bass_label.setText(_translate("Schenky", "Bass:"))
        self.treble_label.setText(_translate("Schenky", "Treble:"))
        self.menuFile.setTitle(_translate("Schenky", "File"))
        self.menuEdit.setTitle(_translate("Schenky", "Edit"))
        self.actionNew.setText(_translate("Schenky", "New"))
        self.actionNew.setStatusTip(_translate("Schenky", "Create a new file"))
        self.actionNew.setShortcut(_translate("Schenky", "Ctrl+N"))
        self.actionOpen.setText(_translate("Schenky", "Open"))
        self.actionOpen.setStatusTip(_translate("Schenky", "Open a file"))
        self.actionOpen.setShortcut(_translate("Schenky", "Ctrl+O"))
        self.actionSave.setText(_translate("Schenky", "Save"))
        self.actionSave.setStatusTip(_translate("Schenky", "Save file"))
        self.actionExit.setText(_translate("Schenky", "Exit"))
        self.actionCopy.setText(_translate("Schenky", "Copy"))
        self.actionCopy.setShortcut(_translate("Schenky", "Ctrl+C"))
        self.actionCut.setText(_translate("Schenky", "Cut"))
        self.actionCut.setShortcut(_translate("Schenky", "Ctrl+X"))
        self.actionPaste.setText(_translate("Schenky", "Paste"))
        self.actionPaste.setShortcut(_translate("Schenky", "Ctrl+V"))
    
    def display_pdf(self):
        d = popplerqt5.Poppler.Document.load("output.pdf")
        if d:
            print("Import successful")
            pdf_page = d.page(0)
            if pdf_page:
                print("Page successful")
                image = pdf_page.renderToImage(72, 72, 0, 0, 741, 191)
                pixmap01 = QtGui.QPixmap.fromImage(image)
                if image.isNull():
                    print("QImage is null")
                else:
                    print("QImage successful")
                pixmap_image = QtGui.QPixmap(pixmap01)
                self.image_label.setPixmap(pixmap_image)
                self.image_label.setAlignment(QtCore.Qt.AlignCenter)
                self.image_label.setScaledContents(True)
                self.image_label.setMinimumSize(1,1)
                self.image_label.show()
                
            else:
                print("Cannot load page")
        else:
            print("Cannot load output.pdf")


if __name__ == "__main__":
    import sys
    app = QtWidgets.QApplication(sys.argv)
    Schenky = QtWidgets.QMainWindow()
    ui = Ui_Schenky()
    ui.setupUi(Schenky)
    Schenky.show()
    ui.display_pdf()
    sys.exit(app.exec_())
